vizro-core/src/vizro/models/_components/table.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from typing import List, Literal
+
+import pandas as pd
+from dash import dash_table, html
+from pydantic import Field, PrivateAttr, validator
+
+import vizro.tables as vt
+from vizro.managers import data_manager
+from vizro.models import Action, VizroBaseModel
+from vizro.models._action._actions_chain import _action_validator_factory
+from vizro.models._components._components_utils import _process_callable_data_frame
+from vizro.models._models_utils import _log_call
+from vizro.models.types import CapturedCallable
+
+logger = logging.getLogger(__name__)
+
+
+class Table(VizroBaseModel):
+    """Wrapper for table components to visualize in dashboard.
+
+    Args:
+        type (Literal["table"]): Defaults to `"table"`.
+        figure (CapturedCallable): Table like object to be displayed. Current choices include:
+            [`dash_table.DataTable`](https://dash.plotly.com/datatable).
+        actions (List[Action]): See [`Action`][vizro.models.Action]. Defaults to `[]`.
+    """
+
+    type: Literal["table"] = "table"
+    figure: CapturedCallable = Field(..., import_path=vt, description="Table to be visualized on dashboard")
+    actions: List[Action] = []
+
+    # Component properties for actions and interactions
+    _output_property: str = PrivateAttr("children")
+
+    # validator
+    set_actions = _action_validator_factory("active_cell")  # type: ignore[pydantic-field]
+    _validate_callable = validator("figure", allow_reuse=True, always=True)(_process_callable_data_frame)
+
+    # Convenience wrapper/syntactic sugar.
+    def __call__(self, **kwargs):
+        kwargs.setdefault("data_frame", data_manager._get_component_data(self.id))  # type: ignore[arg-type]
+        return self.figure(**kwargs)
+
+    # Convenience wrapper/syntactic sugar.
+    def __getitem__(self, arg_name: str):
+        # pydantic discriminated union validation seems to try Table["type"], which throws an error unless we
+        # explicitly redirect it to the correct attribute.
+        if arg_name == "type":
+            return self.type
+        return self.figure[arg_name]
+
+    @_log_call
+    def build(self):
+        return html.Div(dash_table.DataTable(pd.DataFrame().to_dict("records"), []), id=self.id)
```

Comment:
> You probably already discussed this, but I assume there is no way we can insert the self.id into the id of the actual dash datatable right? Currently if we want to use any interactions, we don't reference this self.id, but the id provided in the datatable.

You're right. 

>  Is there any use case where we would reference the self.id instead of the datatable id?

Since `self.id` is used here as the ID of the outer html.Div object, we reference it while calculating Output of the some predefined actions with "children" property.

> We need the className later 

Like it!

Reply:
Do we need to wrap an empty `dash_data.Datatable` into the `dcc.Loading`, similarly as we do with the `vm.Graph` object inside the `Graph.build` method?

Reply:
Hmm.. it depends - does it require a lot of time to load a table ? I would assume that this is actually quite fast as it's just displaying data, rather than compiling a trace. 

I also remember that in the old repository, we had a bug with the datatable flickering, so we had to remove the dcc.Loading wrapper which fixed it. I don't know if this is still the case. 

You can try it out and see if it works and if not, just leave it out

Reply:
1. If the point here is to just generate an empty table then please could we do it in the simplest way possible e.g. `dash_table.DataTable()` or whatever the minimal call here is? No need for `pd.DataFrame()` etc.
2. Actually what is the effect of this existing at all? When do you see the empty table? If it's for page loading purposes do we even need the `dash_table.DataTable()` there at all rather than an empty div? 🤔

Reply:
The dcc.Loading and the empty figure population are two different concepts - the dcc.Loading is just for providing feedback to the user that something is loading 😄  So it's this waiting visual that often is not visible because it's loading fast. So e.g. the dcc.Loading stops when a figure (empty or with data is loaded). 

The empty figure population is for optimizing the performance of actions. As far as I understood, the actions wait for a figure to be loaded to trigger an update of the figure with the default filters/parameters applied. So instead of having to wait for a chart with all data, we just render an empty chart and then the filters/parameters with default selection get applied to render the full chart. Normally you wouldn't see the empty table, unless the actions require lots of time to apply filters etc.

Reply:
Thank you for the explanation @huong-li-nguyen! 🙏

Reply:
1. @antonymilne removed the `pd.DataFrame`, for some reason this didn't work last time I tried, good spot.
2. Thanks for answering this already @huong-li-nguyen !

Reply:
@maxschulz-COL , could you try out the dcc.Loading around the table wrapper then and see if it still works properly? If it doesn't, then just leave it out.